mrmustard/physics/gaussian_integrals.py

```diff
@@ -374,3 +380,92 @@ def complex_gaussian_integral_2(
         b_post = math.zeros((0,), dtype=b.dtype)
 
     return A_post, b_post, c_post
+
+
+def join_Abc_poly(
+    Abc1: Tuple[ComplexMatrix, ComplexVector, ComplexTensor],
+    Abc2: Tuple[ComplexMatrix, ComplexVector, ComplexTensor],
+):
+    r"""Joins two ``(A,b,c)`` triples into a single ``(A,b,c)`` triple by block addition of the ``A``
+    matrices and concatenating the ``b`` vectors.
+
+    Arguments:
+        Abc1: the first ``(A,b,c)`` triple
+        Abc2: the second ``(A,b,c)`` triple
+
+    Returns:
+        The joined ``(A,b,c)`` triple
+    """
+    A1, b1, c1 = Abc1
+    A2, b2, c2 = Abc2
+    c1 = math.astensor(c1)
+    c2 = math.astensor(c2)
+
+    dim_n1 = len(c1.shape)
+    dim_n2 = len(c2.shape)
+
+    dim_m1 = A1.shape[-1] - dim_n1
+    dim_m2 = A2.shape[-1] - dim_n2
+
+    A12 = np.block(
```

Comment:
```suggestion
    A12 = math.block(
```
as well as the `np` calls below